model/auth_test.go

```diff
@@ -92,3 +93,31 @@ func TestAuthDefinitionsStructLevelValidation(t *testing.T) {
 		})
 	}
 }
+
+func TestUnmarshalJSONMultipleAuthProperties(t *testing.T) {
```

Comment:
Can you please add test cases for the OAuth2 scheme too? Many thanks!

Reply:
Will do!